api/src/org/apache/cloudstack/api/command/admin/vpc/VPCOSPFConfigUpdateCmd.java

```diff
@@ -0,0 +1,207 @@
+
+import com.cloud.network.vpc.OSPFZoneConfig;
+import com.cloud.network.vpc.VpcProvisioningService;
+import com.cloud.user.Account;
+import com.cloud.utils.net.cidr.BadCIDRException;
+
+@APICommand(name = "vpcOSPFConfigUpdate", description = "Captures config informaton for ospf zone level params", responseObject = VPCOSPFConfigResponse.class, since = "4.9.0", requestHasSensitiveInfo = true)
+public class VPCOSPFConfigUpdateCmd extends BaseCmd {
+    public static final Logger s_logger = Logger.getLogger(VPCOSPFConfigUpdateCmd.class);
+    private static final String s_name = "vpcospfconfigresponse";
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.ZONE_ID, type = CommandType.UUID, entityType = ZoneResponse.class, required = true, description = "the ID of the Zone")
+    private Long zoneid;
+
+    @Parameter(name = OSPFZoneConfig.s_protocol, type = CommandType.STRING, required = false, description = "the password used to secure inter ospf communication(default ospf)")
+    private String protocol;
+
+    @Parameter(name = OSPFZoneConfig.s_area, type = CommandType.STRING, required = false, description = "Specify the OSPF Area ID")
+    private String area;
+
+    @Parameter(name = OSPFZoneConfig.s_helloInterval, type = CommandType.SHORT, required = false, description = "Set number of seconds for HelloInterval timer value.(default 10)")
+    private Short helloInterval;
+
+    @Parameter(name = OSPFZoneConfig.s_deadInterval, type = CommandType.SHORT, required = false, description = "Set number of seconds for RouterDeadInterval timer value used for Wait Timer and Inactivity Timer.(default 40)")
+    private Short deadInterval;
+
+    @Parameter(name = OSPFZoneConfig.s_retransmitInterval, type = CommandType.SHORT, required = false, description = "Set number of seconds for RxmtInterval timer value.(default 5)")
+    private Short retransmitInterval;
+
+    @Parameter(name = OSPFZoneConfig.s_transitDelay, type = CommandType.SHORT, required = false, description = "Set number of seconds for InfTransDelay value.(default 1)")
+    private Short transitDelay;
+
+    @Parameter(name = OSPFZoneConfig.s_authentication, type = CommandType.STRING, required = false, description = "Md5 or PlainText (default MD5)")
+    private String authentication;
+
+    @Parameter(name = OSPFZoneConfig.s_password, type = CommandType.STRING, required = false, description = "The password used to secure inter quagga communication")
+    private String password;
+
+    @Parameter(name = OSPFZoneConfig.s_superCIDR, type = CommandType.STRING, required = false, description = "The super zone level CIDR for ospf enabled VPCs")
+    private String superCIDR;
+
+    @Parameter(name = OSPFZoneConfig.s_enabled, type = CommandType.BOOLEAN, required = false, description = "The flag to enable or disable ospf for this zone")
```

Comment:
The reasonable default values are provided in setDefault method that is used to bootstrap the values for first time.